src/data/polynomial/coeff.lean

```diff
@@ -86,7 +86,7 @@ add_monoid_algebra.mul_single_zero_apply p a n
 
 lemma coeff_X_pow (k n : ℕ) :
   coeff (X^k : polynomial R) n = if n = k then 1 else 0 :=
-by rw [← monomial_one_eq_X_pow]; simp [monomial, single, eq_comm, coeff]; congr
+by rw [X_pow_eq_monomial]; simp [monomial, single, eq_comm, coeff]; congr
```

Comment:
Let's update the code so it follows modern style conventions:
```suggestion
by { simp [X_pow_eq_monomial, monomial, single, eq_comm, coeff], congr }
```

Reply:
Thanks!

Reply:
Is this a nonterminal `simp` though?

Reply:
`simp, congr` and `simp, refl` are not considered too harmful, right? The reason is probably that there are disagreeing `decidable_eq` instances.

Reply:
Hmm, are `simp, congr` and `simp, refl` less likely to break when new `simp` lemmas are added? (I'm not too concerned about this particular case since it looks like it'd be easy to fix either way.)

Reply:
If `l` is defeq to `r` and a new `simp` lemma turns `l` into something not defeq to `r`, I would say that is a bug in the simp lemma.

Anyway, we can just use `simp only` and avoid the issue altogether (GitHub doesn't allow me to add this as a suggestion):

```lean
by { simp only [X_pow_eq_monomial, monomial, single, eq_comm], congr }
```

Reply:
That's because this conversation is happening on an outdated patch. If you switch to the newer commit, then it would work.

Reply:
:bulb:

Reply:
> If l is defeq to r and a new simp lemma turns l into something not defeq to r, I would say that is a bug in the simp lemma.

Umm... isn't this true of almost every simp lemma? Unless it is a dsimp lemma it's going to rewrite something to something not defeq. So `simp, refl` and `simp, congr` are not safe against new simp lemmas.

Reply:
Sorry, I meant "turns a def equality `l = r` into a non-def equality `l' = r'`. It would imply we are getting further away from confluence.

Reply:
This is unfortunately quite possible, if `r` is a non-idiomatic mess that is defeq to `l` but about which `simp` has nothing to say (which is a common reason to use `refl`).